src/manifolds/ProjectiveSpace.jl

```diff
@@ -0,0 +1,392 @@
+              AbstractEmbeddedManifold{𝔽,DefaultIsometricEmbeddingType} end
+
+@doc raw"""
+    ProjectiveSpace{n,𝔽} <: AbstractProjectiveSpace{𝔽}
+
+The projective space $𝔽ℙ^n$ is the manifold of all lines in $𝔽^{n+1}$.
+The default representation is in the embedding, i.e. as unit norm vectors in
+$𝔽^{n+1}$:
+````math
+𝔽ℙ^n := \bigl\{ p \in 𝔽^{n+1}, p ∼ p z \ \big|\ \lVert p \rVert = 1, z \in 𝔽, |z| = 1 \bigr\},
+````
+where $\sim$ indicates equivalence.
+For example, the real projective space $ℝℙ^n$ is represented as the unit sphere $𝕊^n$, where
+antipodal points are considered equivalent.
+
+The tangent space at point $p$ is given by
+
+````math
+T_p 𝔽ℙ^{n} := \bigl\{ X ∈ 𝔽^{n+1}\ |\ ⟨p,X⟩ = 0 \bigr \},
+````
+
+where $⟨\cdot,\cdot⟩$ denotes the inner product in the embedding $𝔽^{n+1}$.
+
+Note that when $𝔽 = ℍ$, this implementation of $ℍℙ^n$ is the right-quaternionic projective
+space.
+
+# Constructor
+
+    ProjectiveSpace(n[, field=ℝ])
+
+Generate the projective space $𝔽ℙ^{n} ⊂ 𝔽^{n+1}$, defaulting to the real projective space
+$ℝℙ^n$, where `field` can also be used to generate the complex- and right-quaternionic
+projective spaces.
+"""
+struct ProjectiveSpace{N,𝔽} <: AbstractProjectiveSpace{𝔽} end
+ProjectiveSpace(n::Int, field::AbstractNumbers = ℝ) = ProjectiveSpace{n,field}()
+
+@doc raw"""
+    ArrayProjectiveSpace{T<:Tuple,𝔽} <: AbstractProjectiveSpace{𝔽}
+
+The projective space $𝔽ℙ^{n₁,n₂,…,nᵢ}$ is the manifold of all lines in $𝔽^{n₁,n₂,…,nᵢ}$.
+The default representation is in the embedding, i.e. as unit (Frobenius) norm matrices in
+$𝔽^{n₁,n₂,…,nᵢ}$:
+
+````math
+𝔽ℙ^{n_1, n_2, …, n_i} := \bigl\{ p \in 𝔽^{n_1, n_2, …, n_i}, p ∼ p z \ \big|\ \lVert p \rVert = 1, z \in 𝔽, |z| = 1 \bigr\}.
+````
+where $\sim$ indicates equivalence.
+Note that unlike [`ProjectiveSpace`](@ref), the argument for `ArrayProjectiveSpace`
+is given by the size of the embedding.
+This means that [`ProjectiveSpace(2)`](@ref) and `ArrayProjectiveSpace(3)` are the same
+manifold.
+Additionally, `ArrayProjectiveSpace(n,1;field=𝔽)` and [`Grassmann(n,1;field=𝔽)`](@ref) are
+the same.
+
+The tangent space at point $p$ is given by
+
+````math
+T_p 𝔽ℙ^{n_1, n_2, …, n_i} := \bigl\{ X ∈ 𝔽^{n_1, n_2, …, n_i}\ |\ ⟨p,X⟩ = 0 \bigr \},
+````
+
+where $⟨\cdot,\cdot⟩$ denotes the inner product in the embedding $𝔽^{n_1, n_2, …, n_i}$.
+
+# Constructor
+
+    ArrayProjectiveSpace(n₁,n₂,...,nᵢ; field=ℝ)
+
+Generate the projective space $𝔽ℙ^{n_1, n_2, …, n_i}$, defaulting to the real projective
+space, where `field` can also be used to generate the complex- and right-quaternionic
+projective spaces.
+"""
+struct ArrayProjectiveSpace{N,𝔽} <: AbstractProjectiveSpace{𝔽} where {N<:Tuple} end
+function ArrayProjectiveSpace(n::Vararg{Int,I}; field::AbstractNumbers = ℝ) where {I}
+    return ArrayProjectiveSpace{Tuple{n...},field}()
+end
+
+function allocation_promotion_function(::AbstractProjectiveSpace{ℂ}, f, args::Tuple)
+    return complex
```

Comment:
It is tested at https://github.com/JuliaManifolds/Manifolds.jl/blob/d8ef7564ba68cc37ada49afc747af6f7f333c324/test/projective_space.jl#L105, but codecov does not register it for some reason. TBH, I'm not certain what this function does; I copied it from `Grassmann{n,1}`.

Reply:
I know what it does: for a complex manifold the `eltype` of the return value from `exp!` is complex; that's why I think it can be implemented on a `Manifold{ℂ}` - level. But you're right, it sometimes does not get found by codecov, sadly.

Maybe I try that idea later to move this to the supertype.